eliot/tests/test_output.py

```diff
@@ -644,6 +644,29 @@ def test_filedestination_writes_json_bytes(self):
             [message1, message2])
 
 
+    def test_filedestination_flushes(self):
+        """
+        L{FileDestination} flushes after every write, to ensure logs get
+        written out even if the local buffer hasn't filed up.
+        """
+        class CustomException(Exception):
+            pass
+
+        def flush():
+            raise CustomException
+
+        message1 = {"x": 123}
+        bytes_f = BytesIO()
+        bytes_f.flush = flush
```

Comment:
My excuse (and it is an excuse, I didn't think of your much better idea) is that these are file-*like* objects so that's totally legit :grin: But that's just an excuse.